### PR TITLE
move codebase to typescript

### PR DESCRIPTION
**Pull request by HunteRoi**
Switch the entire codebase to TypeScript, update packages and adapt code when necessary. This is the first step to prepare the work of moving to https://github.com/section-ig/igbot

### CODE REVIEW

**Comment by HunteRoi** [comment]
The main work is done. Unfortunately, we are still blocked due to https://github.com/Androz2091/discord-logs/issues/173.
Once a release is done on Androz's side, we should be able to fully test the bot features before merging this PR and officially moving to this new codebase.

File: src/commands/announce.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	adminOnly: true,
+
+	async execute(client: DatadropClient, log: Logger, message: Message, args: string[]) {
+		if (!message.guild) return;
+		const { announce } = config;
+
+		const embed = new EmbedBuilder()
+			.setAuthor({
+				name: 'Annonce',
+				iconURL: 'http://i.imgur.com/zcGyun6.png'
```
Comment (DrakeXorn) [inline src/commands/announce.ts]: Shouldn't it be better to put an HTTPS link ? 😬

**Review by DrakeXorn — DISMISSED** [review]
Seems fine to me 👍 
I was just wondering if we should put newline spaces at the end of the files or not (it's not consistent at the moment)...

**Comment by HunteRoi** [comment]
> Seems fine to me 👍 I was just wondering if we should put newline spaces at the end of the files or not (it's not consistent at the moment)...

Actually, I should reconfigure the linter to do that automatically but haven't given enough time to do it. If you want, you can make commits to this branch to fix it :)

File: src/commands/announce.ts (hunk 1 above)
Comment (HunteRoi) [reply src/commands/announce.ts]:
```suggestion
				iconURL: 'https://i.imgur.com/zcGyun6.png'
```

File: package.json [hunk 2]
```diff
@@ -3,8 +3,10 @@
 	"version": "1.5.0",
 	"main": "datadrop.js",
```
Comment (DrakeXorn) [inline package.json]: Shouldn't the main be changed to `./build/src/datadrop.js` ?

```suggestion
	"main": "./build/src/datadrop.js",
```

**Review by DrakeXorn — COMMENTED** [review]
Seems good to me ! I just want to be sure of something before accepting, and it's related to `package.json`

File: package.json (hunk 2 above)
Comment (HunteRoi) [reply package.json]: It does not really mean anything here. The `main` property defines the entry point of a module when a library is exposed publicly (that is when someone wants to import a module, they would do `require("moduleName")` where moduleName is the name of the main script in the package.json file).
As we are not delivering a module to be used, it does not matter to have a correct value in the main property.

For the sake of it though, I will change it to `"./build/index.js"` which is the real entry point of the bot.
```suggestion
	"main": "./build/index.js",
```

**Pull request merged**